### PR TITLE
add support for java with spring cloud stream framework

### PR DESCRIPTION
**Pull request by rmelian**
- generates a java application using Spring Framework and Spring Cloud Stream to handle the development of event-driven microservices
- for now, always uses RabbitMQ as message broker: Kafka  and others options to come in future PRs

### CODE REVIEW

File: templates/java-spring/src/main/java/com/asyncapi/streetlights/service/MessagePublisher.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.asyncapi.streetlights.service;
```
Comment (fmvilas) [inline templates/java-spring/src/main/java/com/asyncapi/streetlights/service/MessagePublisher.java]: ditto

File: templates/java-spring/src/main/java/com/asyncapi/streetlights/service/MessagePublisher.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import com.asyncapi.streetlights.infrastructure.Bindings;
```
Comment (fmvilas) [inline templates/java-spring/src/main/java/com/asyncapi/streetlights/service/MessagePublisher.java]: ditto

File: templates/java-spring/src/main/resources/application.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+spring:
+  rabbitmq:
```
Comment (fmvilas) [inline templates/java-spring/src/main/resources/application.yml]: Please, note that some people will use `amqp` but not with `RabbitMQ`. Should it be called `amqp` instead?

File: templates/java-spring/src/test/java/com/asyncapi/streetlights/StreetlightsApplicationTests.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package com.asyncapi.streetlights;
```
Comment (fmvilas) [inline templates/java-spring/src/test/java/com/asyncapi/streetlights/StreetlightsApplicationTests.java]: ditto

File: templates/java-spring/src/test/java/com/asyncapi/streetlights/StreetlightsApplicationTests.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class StreetlightsApplicationTests {
```
Comment (fmvilas) [inline templates/java-spring/src/test/java/com/asyncapi/streetlights/StreetlightsApplicationTests.java]: ditto

File: templates/java-spring/.helpers/handlebars.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+//capitalizeFirstLetter
+  Handlebars.registerHelper('capFirst', (str) => {
+    return str.charAt(0).toUpperCase() + str.slice(1);
+  });
+
+//convert string to java constant format: lightMeasuredPublish -> LIGHT_MEASURED_PUBLISH
+  Handlebars.registerHelper('javaConst', (str) => {
+    var tokenList = str.split(/(?=[A-Z])/);
+    return tokenList.map(function(x){ return x.toUpperCase() }).join("_");
+  });
```
[the remark this answers is not in the document]
Comment (rmelian) [reply templates/java-spring/.helpers/handlebars.js]: as a convention, the Java community uses UPPER_CASE separated by underscore

[the remark this answers is not in the document]
Comment (rmelian) [reply templates/java-spring/.helpers/handlebars.js]: ohh wait, I got it thanks

**Comment by rmelian** [comment]
sure, will do

**Comment by rmelian** [comment]
readme added :)

**Pull request merged**